### PR TITLE
Fixing issue with Slack API and DM in Riker

### PR DESCRIPTION
**Pull request by CSdread**
Issue when code testing for DM vs Channel broke due to API change. This code can be improved with link to stackoverflow in the code but still not perfect.

### CODE REVIEW

**Comment by joemiller** [comment]
👍

**Comment by joemiller** [comment]
+1

File: pkg/chat/slackbot/slack.go [hunk 1]
```diff
@@ -245,6 +253,29 @@ func (b *SlackBot) Run() {
...
+func (b *SlackBot) isChan(ev *slack.MessageEvent) (bool, error) {
+	//  we need to detect  a direct message from a channel message, and unfortunately slack doens't make that super awesome
+	// https://stackoverflow.com/questions/41111227/how-can-a-slack-bot-detect-a-direct-message-vs-a-message-in-a-channel
+	b.Lock()
+	isChan, ok := b.channels[ev.Channel]
+	if !ok {
+		_, err := b.rtm.GetChannelInfo(ev.Channel)
+		if err != nil && (err.Error() != "channel_not_found" && err.Error() != "method_not_supported_for_channel_type") {
+			b.log.Debug("not dm not channel: ", err)
+			b.Unlock()
```
Comment (spheromak) [inline pkg/chat/slackbot/slack.go]: since it's a method now. the unlock may  be able to be Defered to prevent issues in the future.

**Comment by spheromak** [comment]
+1 in general  prob should defer the unlock to make it cleaner.  If you're up for it a pass at CI for circle2 and moving to go modules on another PR  would be super awesome.

**Comment by spheromak** [comment]
thinking about the error/ error messages. Does the Library return any typed error yet here ? vs string matching. Also perhaps theres a Const for the Err message in the upstream we can match against instead of replicating that string here.

File: pkg/chat/slackbot/slack.go (hunk 1 above)
Comment (CSdread) [reply pkg/chat/slackbot/slack.go]: oh good call, have not been in go land for a while

**Comment by CSdread** [comment]
this does not return typed error message

**Pull request merged**